libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -513,12 +509,12 @@ public String getNewAuthToken() {
             }
 
             // Invalidate current auth token.
-            final String cachedAuthToken = clientManager.accountManager.peekAuthToken(acc, AccountManager.KEY_AUTHTOKEN);
+            final String cachedAuthToken = clientManager.peekRestClient(acc).getAuthToken();
             clientManager.invalidateToken(cachedAuthToken);
             String newAuthToken = null;
             String newInstanceUrl = null;
             try {
-                final Bundle bundle = clientManager.accountManager.getAuthToken(acc, AccountManager.KEY_AUTHTOKEN, null, false, null, null).getResult();
+                final Bundle bundle = refreshStaleToken(acc);
```

Comment:
Refresh the token internally instead of calling `getAuthToken()` on `AccountManager`.

Reply:
Do we need the refresh code [here](https://github.com/bhariharan/SalesforceMobileSDK-Android/blob/authenticator/libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/AuthenticatorService.java#L185) ?
Can we avoid the duplication?

Reply:
Yes, we need that too, for external modules that might use our authenticator to auth using Salesforce. I could extract it into a util class but I didn't want to expose it in a public method. I'll try to organize this better when I do the IDP flow on Android.